src/v/cluster/self_test/cloudcheck.cc

```diff
@@ -214,7 +223,7 @@ ss::future<self_test_result> cloudcheck::verify_upload(
   cloud_storage_clients::object_key key,
   const std::optional<iobuf>& payload) {
     auto result = self_test_result{
-      .name = _opts.name, .info = "upload", .test_type = "cloud_storage"};
+      .name = _opts.name, .info = "Put", .test_type = "cloud_storage"};
```

Comment:
This commit seems like it could be split up. Stuff that it looks like it's doing:

1. Renaming things (e.g. operation names upload vs Put)
2. Fixing issues with exception / error handling in try/catch blocks?
3. Changes to max_keys parameter handling
4. Verify head
5. Verify deletes

Reply:
Thanks for the feedback, I have split this commit up into several commits.